From b179207fd09b3c307ac007ca7470aae12f12a2bd Mon Sep 17 00:00:00 2001
From: wangrq <dlazycat@gmail.com>
Date: Sun, 30 Nov 2014 16:23:21 +0100
Subject: [PATCH] test tile edge

---
 game/game.lua | 34 ++++++++++++++++++++++++----------
 1 file changed, 24 insertions(+), 10 deletions(-)

diff --git a/game/game.lua b/game/game.lua
index 1600298..853e15a 100644
--- a/game/game.lua
+++ b/game/game.lua
@@ -368,16 +368,30 @@ function draw_tiles()
     v = Level.tiles[k]
     -- This code can't run properly on the box because the difference
     -- of screen:copyfrom function . Wait for further improvement
-    if v.x-gameCounter+v.width>0 and v.visibility==true and v.x-gameCounter+v.width<w + v.width then
-      if v.name == "obstacle3" then
-        move_cloud(v)
-      elseif v.name == "obstacle4" then
-        move_flame(v)
-      end
-      if v.image == nil then
-        print("The tile the want to draw == nil")
-        v.image = gfx.loadpng("images/font/Z.png")
-      end
+    if v.name == "obstacle3" then
+      move_cloud(v)
+    elseif v.name == "obstacle4" then
+      move_flame(v)
+    end
+    if v.image == nil then
+      print("The tile the want to draw == nil")
+      v.image = gfx.loadpng("images/font/Z.png")
+    end
+    if v.x-gameCounter<0 and v.visibility==true then
+      
+      --[[
+      local x0 = (gameCounter - v.x)
+      local w0 = (v.x+v.width-gameCounter)
+      ]]
+      local x0 = v.x-gameCounter
+      local w0 = v.width
+
+      screen:copyfrom(v.image, {x=x0, y=v.y, width=w0, height=v.height}, {x=0, y=v.y, width=w0, height=v.height},true)
+
+    elseif v.x+v.width > w+gameCounter and v.visibility==true then
+      screen:copyfrom(v.image,{x=0, y=v.y, width=w+gameCounter-v.x, height=v.height}
+        ,{x=w+gameCounter-v.x, y=v.y, width=w+gameCounter-v.x, height=v.height},true)
+    elseif v.visibility == true then
       screen:copyfrom(v.image,nil,{x=v.x-gameCounter,y=v.y,width=v.width,height=v.height},true)
     end
   end
-- 
1.9.0.msysgit.0

